jerry-core/jerry.cpp

```diff
@@ -430,6 +431,104 @@ jerry_api_create_object (void)
 } /* jerry_api_create_object */
 
 /**
+ * Create an array object
+ *
+ * Note:
+ *      caller should release the object with jerry_api_release_object, just when the value becomes unnecessary.
+ *
+ * @return pointer to created array object
+ */
+jerry_api_object_t *
+jerry_api_create_array_object (jerry_api_size_t size) /* size of array */
+{
+  JERRY_ASSERT (size > 0);
+
+  ecma_number_t *length_num_p = ecma_alloc_number ();
+  *length_num_p = ecma_uint32_to_number (size);
+  ecma_value_t array_length = ecma_make_number_value (length_num_p);
+
+  jerry_api_length_t argument_size = 1;
+  ecma_completion_value_t get_completion = ecma_op_create_array_object (&array_length, argument_size, true);
+  ecma_value_t val = ecma_get_completion_value_value (get_completion);
+  jerry_api_object_t *obj_p = ecma_get_object_from_value (val);
+
+  if (!ecma_is_completion_value_normal (get_completion))
+  {
+    JERRY_ASSERT (ecma_is_completion_value_throw (get_completion));
```

Comment:
Exception couldn't be thrown by `ecma_op_create_array_object`, so this `if` block could be replaced with assertion like `JERRY_ASSERT (ecma_is_completion_value_normal (new_array_completion));` before line 452.